esp/src/src-react/components/Logs.tsx

```diff
@@ -13,6 +13,8 @@ import { Filter } from "./forms/Filter";
 import { Fields } from "./forms/Fields";
 import { ShortVerticalDivider } from "./Common";
 
+const defaultStartDate = new Date(new Date().getTime() - 8 * 3600 * 1000);
+
```

Comment:
I think this needs to be moved into the component otherwise it may be invalid if ECL Watch is left open for a long period of time?

Reply:
defaultStartDate is calculated the first time this .js file loaded, not when the component is displayed?

Reply:
Yes, I wasn't able to get that to work correctly as a memoized const defined in the component. So I'd added the check in the query useMemo so that if the StartDate is outside of a window of time and adjusts (https://github.com/hpcc-systems/HPCC-Platform/pull/17335/files#diff-99824c7f4929c5f126612e069e2de6e70c53dae6b40d6208cdbade96570f59e8R99).